src/ProjectModule/Components/ProjectList/ProjectList.tsx

```diff
@@ -1,7 +1,122 @@
-import React from 'react'
+import { useEffect, useState } from "react";
+import Modal from 'react-bootstrap/Modal';
+import Dropdown from 'react-bootstrap/Dropdown';
+import axios from "axios";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import styles from "./ProjectList.module.css"
+import { useForm } from "react-hook-form";
+import noData from "../../../assets/noData.jpg"
 
 export default function ProjectList() {
+const[projects,setProjects]=useState([]);
+
+const {register,handleSubmit,setValue,formState:{errors}}=useForm();
+const[pageArray,setPageArray]=useState([]);
+
+const getAllProject=async(pageNo:number,pageSize:number,title:string)=>{
```

Comment:
I suggest taking the parameters as a single object, I prefer that for multiple reasons:
1-  We don't need to remember the order of the params as we can write them as:
```
{pageNumber:15; 
 pageSize:20;
title:"myProject";
}

or

{
title:"myProject";
pageNumber:15; 
 pageSize:20;
}
```
2- It also make the code more readable, as we know exactly what each number reprsent, currently when we call the function it looks something like `getAllProjects(1,5,"myTitle")`  and we need to see the function definition to see what each value represents.

Reply:
We can also add default values for the function for the page number and page size and title, so that we can call `getProjects()` without needing to provide the values for all parameters, since they should be optional not required.